Classes/Updates/FieldLogoMigrationWizard.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the package jweiland/daycarecenters.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace JWeiland\Daycarecenters\Updates;
+
+use Doctrine\DBAL\DBALException;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use TYPO3\CMS\Core\Core\Environment;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\Database\Query\Expression\ExpressionBuilder;
+use TYPO3\CMS\Core\Resource\FileInterface;
+use TYPO3\CMS\Core\Resource\ResourceStorage;
+use TYPO3\CMS\Core\Resource\StorageRepository;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Install\Attribute\UpgradeWizard;
+use TYPO3\CMS\Install\Updates\UpgradeWizardInterface;
+
+#[UpgradeWizard('daycarecentersFieldLogoMigration')]
+class FieldLogoMigrationWizard implements UpgradeWizardInterface, LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    public function getTitle(): string
+    {
+        return '[daycarecenters] Migrate FAL relations where field name with logo to logos';
+    }
+
+    public function getDescription(): string
+    {
+        return 'From version 6.0 the field logo renamed to logos';
+    }
+
+    public function updateNecessary(): bool
+    {
+        return $this->getRecordsFromTable();
+    }
+
+    public function getPrerequisites(): array
+    {
+        return [];
+    }
+
+    public function executeUpdate(): bool
+    {
+        $customMessage = '';
+
+        try {
+            $storages = GeneralUtility::makeInstance(StorageRepository::class)->findAll();
+            $this->storage = $storages[0];
+            $records = $this->getRecordsFromTable();
+            foreach ($records as $record) {
+                $this->migrateField($record, $customMessage, $dbQueries);
+            }
+        } catch (\Exception $e) {
+            $customMessage .= PHP_EOL . $e->getMessage();
+        }
+
+        return empty($customMessage);
+    }
+
+    /**
+     * Get records from table where the field to migrate is not empty (NOT NULL and != '')
+     * and also not numeric (which means that it is migrated)
+     *
+     * @throws \RuntimeException
+     */
+    protected function getRecordsFromTable(): bool
+    {
+        $connectionPool = GeneralUtility::makeInstance(ConnectionPool::class);
+        $queryBuilder = $connectionPool->getQueryBuilderForTable('sys_file_reference');
+        $queryBuilder->getRestrictions()->removeAll();
+
+        try {
+            $result = $queryBuilder
+                ->count('uid')
+                ->from('sys_file_reference')
+                ->where(
+                    $queryBuilder->expr()->eq(
+                        'fieldname',
+                        $queryBuilder->createNamedParameter('logo', \PDO::PARAM_STR)
+                    )
+                )
+                ->orWhere(
```

Comment:
The query is wrong. It will look like:

```
SELECT COUNT(*)
FROM `sys_file_reference`
WHERE (`fieldname` = 'logo')
OR (`tablenames` = 'tx_daycarecenters_domain_model_kita')
OR (`tablenames` = 'tx_daycarecenters_domain_model_holder')
```

I prefer using `$queryBuilder->expr->or()` within the `where` part.